Aktualizace

janpecha opened the pull request:
* zapnuto zvýrazňování syntaxe
* aktualizován changelog
* přidána stránka Mapper
* přidána stránka Ukázky použití a zajímavá řešení + CamelCase to under_score mapper (přesunuto z wiki https://github.com/Tharos/LeanMapper/wiki/camelcase-to-under-score-mapper)
* přidána stránka EntityFactory
* přidána stránka Integrace Lean Mapper do aplikace (přesunuto z wiki https://github.com/Tharos/LeanMapper/wiki/integrace-lean-mapperu-do-aplikace)

janpecha committed:
CSS: enabled syntax-highlighting

janpecha committed:
Changelog: updated

janpecha committed:
Docs: added Mapper

janpecha committed:
Added "Ukazky pouziti a zajimava reseni"

janpecha committed:
Tutorials: added "CamelCase to under_score mapper"

Moved from https://github.com/Tharos/LeanMapper/wiki/camelcase-to-under-score-mapper

janpecha committed:
Docs: added EntityFactory

castamir commented on cs/docs/integrace-do-aplikace.md:
```diff
@@ -0,0 +1,114 @@
+    # registrace Lean Mapperu
+    - LeanMapper\Connection(%leanmapper%)
+    - LeanMapper\DefaultMapper
+    - LeanMapper\DefaultEntityFactory
```
pokud si dobre vzpominam, tak tohle neni potreba definovat, existuje DI\Extension pro nette a dokonce regne i repozitare viz https://github.com/Tharos/LeanMapper/blob/develop/src/LeanMapper/Bridges/Nette/DI/LeanMapperExtension.php

janpecha replied:
Rozšíření pro Nette nepoužívám, ale ano, i to je jedna z cest. Ideální by asi bylo zmínit oboje - DI rozšíření jako doporučenou cestu a tohle nastavení jako alternativu pro ty, kteří nechtějí nemůžou DI rozšíření použít.

janpecha replied:
Přidal jsem část o použití DI rozšíření (snad je použito správně).

castamir commented on cs/docs/integrace-do-aplikace.md:
```diff
@@ -0,0 +1,114 @@
+
+V rámci presenteru pak máme repositář přístupný přes `$this->bookRepository`.
+
+**Tip:** pokud nechcete, nebo nemůžete použít anotaci `@inject`, lze pro předání repositáře použít constructor nebo [`inject*()` metodu](http://pla.nette.org/cs/inject-autowire).
```
Tohle bych napsal naopak. Konstruktor je jedine spravne reseni predani zavislosti. Injectovani, at uz pres anotace nebo pres metodu, je jen vseobecne pouzivana berlicka specialne pro presentery, aby se vyvojari vyhnuli problemu "contructor hell" pri velke dedicnosti, ktera je u presenteru bezna. V modelu aplikace (ve smyslu MVC) je nutne predat zavislosti vyhradne pres konstruktor.

janpecha replied:
Jasně, že ano. Psal jsem to hlavně s ohledem na začátečníky, kteří se snadno do contructor hellu v presenteru zapletou a jeho vyřešení je pro ně obtížné, pokročilejší uživatelé tohle nepotřebují, ti si to rovnou předají přes kontruktor.

janpecha replied:
Přidal jsem předání přes konstruktor.

castamir commented on cs/tutorials/camelcase-to-underscore-mapper.md:
```diff
@@ -0,0 +1,62 @@
+
+Existuje mnoho důvodů proč v databázi pojmenovávat *sloupce/tabulky* pouze malými písmeny a jednotlivá slova oddělovat podtržítkem `_` *(tzv. under score konvence)*. Pokud bychom ale na straně PHP chtěli k těmto datům přistupovat pomocí *camel case konvence*, tedy vizuálně oddělovat velkým písmenem na začátku jednotlivých slov, **LeanMapper** k tomu nabízí všechny potřebné prostředky.
+
+K zprovoznění této funkcionality postačí upravit [**mapper**](/cs/docs/mapper/), tedy třídu, která dědí po `LeanMapper\DefaultMapper` nebo jen implementuje rozhraní `LeanMapper\IMapper`. Nejprve si připravíme funkce, které nám budou převádět `fooBar` na `foo_bar` a obráceně. Můžeme je klidně implementovat jako statické veřejné metody vlastního mapperu.
```
Striktne vzato, metody pro prevod nazvu z jedne konvence do jine nepatri do Mapperu, ale jinam, kvuli pouzitelnosti mimo LeanMapper. Jejich implementace muze zustat staticka (stejne jako trida Strings z Nette) a v Mapperu to jen pouzit.

janpecha replied:
Striktně vzato asi ano, ale pokud nikde jinde ty metody nepotřebujeme, nedává jejich vyčlenění do další třídy smysl. Navíc tohle je jen příklad, je asi zbytečné to komplikovat dalšími třídami.

castamir reviewed (changes requested):
prosim o vyjadreni pripadne o upravy

castamir commented:
Jinak vyborna prace! ;-)

castamir commented on cs/docs/integrace-do-aplikace.md:
```diff
@@ -0,0 +1,145 @@
+
+leanmapper:
+    db:
+        host: localhost
```
přidej sem ještě driver, ať je to stejné jako dole

janpecha replied:
hotovo.

PS. přidal jsem ještě commit, který aktualizuje rok v patičce (z 2013-2016 na 2013-2017).

janpecha committed:
Docs: added Integrace Lean Mapper do aplikace

Moved from https://github.com/Tharos/LeanMapper/wiki/integrace-lean-mapperu-do-aplikace

janpecha committed:
Footer: updated year

castamir commented:
great job ;)
